docs/vgclog_examples.md

```diff
@@ -0,0 +1,594 @@
+|concurrent mark      |intermediate |concurrent                     | `<gc-start>`, `<gc-end>`     |`<concurrent-trace-info>` records the progress of the concurrent mark increment|
+|final collection     |final        | STW              | `<concurrent-global-final>`  |Triggered when the card-cleaning threshold is reached. Child element is `<concurrent-trace-info reason=””>`. Operations and sub-operations include a final concurrent mark, a sweep, and an optional class unload and compact.|
+
+The global GC cycle follows a general structure in the verbose GC log as shown. The lines are indented to help illustrate the flow and attributes and some child elements are omitted for clarity. The structure includes a partial GC cycle that starts and finishes within the global cycle:
+
+```xml
+
+<concurrent-kickoff/>       (global cycle 1st increment recorded)
+
+<exclusive-start/>          (STW pause starts)
+
+<cycle-start/>              (global cycle starts)
+
+<exclusive-end/>            (STW pause ends)
+
+
+(global cycle 2nd increment running concurrently)
+
+
+<exclusive-start/>          (STW for partial GC cycle starts)
+
+
+...                         (partial GC cycle starts and completes)
+
+
+<exclusive-end/>            (STW for partial GC cycle ends)
+
+<exclusive-start/>          (STW pause starts)
+
+<concurrent-global-final/>  (global cycle final increment recorded)
+
+<gc-start/>                 (global cycle final increment starts)
+
+<allocation-stats/>
+
+<mem-info>                  (memory status before operations)
+
+<mem></mem>                 (status of different types of memory)
+
+</mem-info>
+
+</gc-start>
+
+<gc-op> “type=rs-scan"</gc-op>        (roots scanning completed)
+
+<gc-op>”type=card-cleaning" </gc-op>  (card cleaning completed)
+
+<gc-op> “type=mark”</gc-op>           (mark operation completed)
+
+<gc-op> “type=classunload”</gc-op>    (class unload operation completed)
+
+<gc-op ”type=sweep” />                (sweep operation completed)
+
+<gc-end>                     (global cycle final increment ends)
+
+<mem-info>                   (memory status after operations)
+
+<mem></mem>                  (status of different types of memory)
+
+</mem-info>
+
+</gc-end> 
+
+</cycle-end>                 (global cycle ends)
+
+<exclusive-end>              (STW pause ends)
+
+<exclusive-start>            (STW pause starts)
+...
+
+```
+
+The first activity in the cycle is recorded by a `<concurrent-kickoff>` element, which records the start of the first of three increments that make up a `gencon` Global GC cycle. The `<concurrent-kickoff>` element records:
+
+- the reason why the GC cycle was triggered. For a `gencon` global cycle, the cycle is triggered when the amount of available memory reaches a particular threshold value.
+- The target number of bytes the cycle aims to reclaim in the heap.
+- The current available memory in the different parts of the heap.
```

Comment:
This is correct, but we also report the treshold too: since remainingFree is just slightly below thresholdFreeBytes, we are kicking off this global.

Reply:
- the reason why the GC cycle was triggered. For a `gencon` global cycle, the cycle is triggered when the amount of free memory has decreased to a threshold value, the `thresholdFreeBytes` value. 
- The target number of bytes, `targetBytes`, that the cycle aims to mark concurrently
- The current available memory in the different parts of the heap.

```xml
<concurrent-kickoff id="12362" timestamp="2020-10-18T13:35:44.341">

<kickoff reason="threshold reached" targetBytes="239014924" thresholdFreeBytes="33024922" remainingFree="32933776" tenureFreeBytes="42439200" nurseryFreeBytes="32933776" />

</concurrent-kickoff>
```

For this example, the `remainingFree` bytes value of 32.9MB (32933776) is approaching the `thresholdFreeBytes` value of 33.0MB (33024922) so a global cycle is triggered.